sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -421,4 +430,15 @@ export class JsonRpcProvider extends Provider {
       );
     }
   }
+
+  async subscribeEvent(
+    filter: SuiEventFilter,
```

Comment:
What is the desired behavior if filter == `{All: []}`, will it get subscribed all the events in the Network? cc @longbowlu